Source/Core/Common/x64Emitter.h

```diff
@@ -1122,6 +1129,15 @@ class XEmitter
     ABI_CallFunction(func);
   }
 
+  // Pass a pointer and register as a parameter.
+  template <typename FunctionPointer>
+  void ABI_CallFunctionPR(FunctionPointer func, const void* ptr, X64Reg reg1)
+  {
+    MOV(64, R(ABI_PARAM1), Imm64(reinterpret_cast<u64>(ptr)));
+    MOV(32, R(ABI_PARAM2), R(reg1));
```

Comment:
I'm still not sure this is right. Consider the following:

ABI_PARAM1 = rcx
ABI_PARAM2 = rdx
reg1 = rcx

which would result in

(outside this function)
mov ecx, (parameter value)
(inside this function)
mov rcx, (function address)
mov edx, ecx

Clearly not what we want! Unless I'm grossly misunderstanding something here? If you just reverse the order of the two instructions it would work, though.